core/src/main/java/org/openstack4j/openstack/artifact/internal/BaseArtifactServiceImpl.java

```diff
@@ -36,27 +37,27 @@ protected BaseArtifactServiceImpl(ArtifactType artifactType) {
 
     protected <T> T list(Class<T> clazz) {
 
-        return get(clazz,uri("/artifacts/"+ artifactType.value())).execute();
+        return get(clazz,uri(ClientConstants.PATH_ARTIFACTS+ClientConstants.URI_SEP+ artifactType.value())).execute();
 
     }
 
     protected <T> T get(String artifactId, Class<T> clazz) {
-        return get(clazz,uri("/artifacts/"+ artifactType.value()+"/%s",artifactId)).execute();
+        return get(clazz,uri(ClientConstants.PATH_ARTIFACTS+ClientConstants.URI_SEP+ artifactType.value()+"/%s",artifactId)).execute();
```

Comment:
checkNotNull() is called in specific implementation class ToscaTemplatesArtifactServiceImpl.java.